src/server/session.ts

```diff
@@ -380,7 +380,7 @@ namespace ts.server {
         // correct results to all other projects.
 
         const defaultProjectResults = perProjectResults.get(defaultProject)!;
-        if (defaultProjectResults[0].references[0].isDefinition === undefined) {
+        if (defaultProjectResults?.[0].references[0]?.isDefinition === undefined) {

```

Comment:
If there are no results at all, `defaultProjectResults` will be undefined.  If, as for some string literals, there's a definition but no references, `references[0]` will be undefined.  Both are expected and clearing the `isDefinition` flag is (trivially) correct for both.

Reply:
```suggestion
        const defaultProjectResults = perProjectResults.get(defaultProject);
        if (defaultProjectResults?.[0].references[0]?.isDefinition === undefined) {        
```

Reply:
🤦‍♂️

Reply:
Applying your fix directly doubled up the declaration.  I've force-pushed a clean commit.